Application/EdFi.Ods.Features/ChangeQueries/ChangeVersionFilterMappingNHibernateConfigurationActivity.cs

```diff
@@ -47,7 +49,9 @@ public void Execute(object sender, BindMappingEventArgs e)
                                name = "long"
                            },
                     notnull = true,
-                    generated = HbmPropertyGeneration.Always
+                    generated = HbmPropertyGeneration.Never,
```

Comment:
See the next generated insert statements.

**Before the change:**
`exec sp_executesql N'INSERT INTO edfi.Program (LastModifiedDate, CreateDate, Id, ProgramId, CreatedByOwnershipTokenId, EducationOrganizationId, ProgramName, ProgramTypeDescriptorId) VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6, @p7)',N'@p0 datetime2(7),@p1 datetime2(7),@p2 uniqueidentifier,@p3 nvarchar(20),@p4 smallint,@p5 int,@p6 nvarchar(60),@p7 int',@p0='2024-09-23 18:20:53.4126335',@p1='2024-09-23 18:20:53.4131996',@p2='BDC38C82-0C7B-4E8E-A9AD-266DB8323E87',@p3=N'e5115af4c859438e9d98',@p4=64,@p5=255901,@p6=N'e25fb361b98b467290b05f709cf4eacf',@p7=1354`

**After the change:**
`exec sp_executesql N'INSERT INTO edfi.Program (LastModifiedDate, CreateDate, Id, ProgramId, ChangeVersion, CreatedByOwnershipTokenId, EducationOrganizationId, ProgramName, ProgramTypeDescriptorId) VALUES (@p0, @p1, @p2, @p3, @p4, @p5, @p6, @p7, @p8)',N'@p0 datetime2(7),@p1 datetime2(7),@p2 uniqueidentifier,@p3 nvarchar(20),@p4 bigint,@p5 smallint,@p6 int,@p7 nvarchar(60),@p8 int',@p0='2024-09-23 18:18:42.6179315',@p1='2024-09-23 18:18:42.6182858',@p2='DDB231AD-B640-49CB-86D3-A3ADBB632DA0',@p3=N'0ae0ba2b3383460b8c58',@p4=0,@p5=57,@p6=255901,@p7=N'b9c3a85b9c964e9c909576f62c388853',@p8=1354`

Notice that after the change, the `ChangeVersion` column gets specified with the value of `0`  (so the `ChangeVersionSequence` is no longer used), meaning that newly created resources all have `ChangeVersion=0`.

Reply:
It looks like there's another property that must be set to get NHibernate to respect the insert/update settings in the code-based mappings. @vimayya We probably should review and consider applying the same change to the Ownership-based authorization strategy mapping.

https://github.com/nhibernate/nhibernate-core/blob/894529983457c110f2934904f27e2c3e9398c581/src/NHibernate/Cfg/XmlHbmBinding/PropertiesBinder.cs#L262